src/character.cpp

```diff
@@ -4860,6 +4860,7 @@ void Character::check_needs_extremes()
 
     // Sleep deprivation kicks in if lack of sleep is avoided with stimulants or otherwise for long periods of time
     int sleep_deprivation = get_sleep_deprivation();
+    bool sleep_deprivation_toggle = true;
```

Comment:
I think you may also want this to be a variable stored in the character, because `sleep_deprivation_toggle` is scoped to only the function it's inside, and as a result, `sleep_deprivation_toggle` would be `true` every time the function is called, meaning that when it gets to this segment:
```
if( sleep_deprivation_toggle ) {
    mod_daily_health( -1, -200 );
    sleep_deprivation_toggle = false;
} else {
    sleep_deprivation_toggle = true;
}
```
it would decrease `daily_health` every hour instead of every 2 hours as intended.